console/frontend/src/main/java/nl/nn/adapterframework/console/ConsoleFrontend.java

```diff
@@ -0,0 +1,139 @@
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import nl.nn.adapterframework.lifecycle.DynamicRegistration;
+
+/**
+ * Contains the component annotation so it will be picked up by the core (SpringEnvironmentContext.xml).
+ * When using Spring boot (and the FrankConsoleContext.xml), where classpath scanning is enabled, 
+ * it will directly try to configure this servlet without any configuration.
+ * Hence it should use the RegisterServletEndpoints which uses a ServletConfiguration to initialize the Servlet.
+ * 
+ * @author Niels Meijer
+ */
+@Log4j2
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
+public class ConsoleFrontend extends HttpServlet implements DynamicRegistration.Servlet, EnvironmentAware, InitializingBean {
+
+	private static final long serialVersionUID = 1L;
+	private static final String WELCOME_FILE = "/index.html";
+	private static final String DEFAULT_CONSOLE_PATH = "classpath:/console";
+
+	private transient String frontendPath = null;
+	@Setter private transient Environment environment;
+
+	@Override
+	public void afterPropertiesSet() {
+		if(environment != null && Arrays.asList(environment.getActiveProfiles()).contains("dev")) {
+			String devFrontendLocation = environment.getProperty("frontend.resources.location");
+			if(devFrontendLocation == null) {
+				Path rootPath = Paths.get("").toAbsolutePath(); // get default location based on current working directory, in IntelliJ this is the project root.
+				devFrontendLocation = rootPath.resolve("console/frontend/target/frontend/").toString(); //Navigate to the target of the frontend module
+			}
+
+			frontendPath = ResourceUtils.FILE_URL_PREFIX + FilenameUtils.getFullPath(devFrontendLocation);
+			log.info("found frontend path [{}]", frontendPath);
+		}
+
+		frontendPath = DEFAULT_CONSOLE_PATH;
+	}
+
+	@Override
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		String path = req.getPathInfo();
+		if(StringUtils.isBlank(path)) { //getPathInfo may return null, redirect to {base}+'/' when that happens.
+			String fullPath = req.getRequestURI();
+			if(StringUtils.isNotBlank(req.getContextPath())) {
+				fullPath = fullPath.substring(req.getContextPath().length());
+			}
+			System.err.println(fullPath);
+//			String redirectUrl = req.getRequestURL().append("/").toString();
+//			if(!fullPath.equals("/")) {
+//				resp.sendRedirect(fullPath);
+//				return;
+//			}
+			path = fullPath;
+		}
+		if(path.equals("/")) {
+			path += WELCOME_FILE;
+		}
+
+		URL resource;
+		try {
+			resource = new URL(frontendPath+path);
+			log.info("looked up [{}]", resource);
+		} catch (IOException e) {
+			log.warn("unable to locate file [{}]", path, e);
+			resp.sendError(404, "file not found");
+			return;
+		}
+
+		String mimeType = getServletContext().getMimeType(path);
+		resp.setContentType(mimeType != null ? mimeType : "application/octet-stream");
+
+		try(InputStream in = resource.openStream()) {
```

Comment:
@nielsm5, perhaps add validation to the given `path` var, as early in this method as possible, to validate only path elements/chars are inside `path` and no trickery or hacking is going on. E.g. no protocol change or patterns like '../../' are allowed.

Reply:
I don't see yet how this could harm us here but perhaps we could do some validation on the input? That we cannot serve any response that we do not want to serve.

Reply:
I can validate/normalize the input, but the url will be decoded by the servlet..